wicketd/src/update_events.rs

```diff
@@ -22,6 +27,13 @@ pub struct UpdateState {
 #[derive(Clone, Debug, JsonSchema, Serialize)]
 #[serde(rename_all = "snake_case", tag = "state", content = "data")]
 pub enum UpdateStateKind {
+    // This is a hack to handle cases where a completion has just happened and
+    // there's no progress from the next step yet: see
+    // https://github.com/oxidecomputer/omicron/pull/2464#discussion_r1123308999
+    // for some discussion.
```

Comment:
Thanks for working through this. I know where we landed doesn't feel _super_ satisfying but I think it's perfectly reasonable.